src/test/bip324_tests.cpp

```diff
@@ -89,9 +92,67 @@ void TestBIP324PacketVector(
     BOOST_CHECK(ciphertext.size() >= out_ciphertext_endswith.size());
     BOOST_CHECK(Span{out_ciphertext_endswith} == Span{ciphertext}.last(out_ciphertext_endswith.size()));
 
-    // Note that we don't test decryption here, as the test vectors don't provide the other party's
-    // private key, so we cannot act like them. See the bip324_cipher_roundtrip fuzz test for a test
-    // that does cover decryption.
+    for (unsigned error = 0; error <= 11; ++error) {
```

Comment:
off-by-one, the "message index wrong" kind of error type (error=12) is currently not tested:
```suggestion
    for (unsigned error = 0; error <= 12; ++error) {
```
(to avoid issues like this, could add an enum with error types and give the highest error sth like ERROR_HIGHEST, to be used in the loop counter, but that's probably overkill)